packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -199,7 +396,10 @@ class IOSDevice extends Device {
 
   final DarwinArch cpuArchitecture;
 
-  final IOSDeviceConnectionInterface interfaceType;
+  IOSDeviceConnectionInterface interfaceType;
```

Comment:
So this is both known at the time we construct the class (that is, non-null) but also mutable (non-final)?

Reply:
Yeah, so when we get the initial IOSDeviceConnectionInterface, it might be incorrect. For example, `xcdevice list` will say a wireless device has an interface of `usb` (it will also have an error) if it didn't have long enough to check for the network connection.

Here's an example of what it will return for a wireless device it didn't have enough time to properly check:
```json
  {
    "modelCode" : "iPad14,3",
    "simulator" : false,
    "modelName" : "iPad Pro (11-inch) (4th generation)\"",
    "error" : {
      "code" : -13,
      "failureReason" : "",
      "underlyingErrors" : [
        {
          "code" : 4,
          "failureReason" : "",
          "description" : "iPad is locked.",
          "recoverySuggestion" : "To use iPad with Xcode, unlock it.",
          "domain" : "DVTDeviceIneligibilityErrorDomain"
        }
      ],
      "description" : "iPad is not connected",
      "recoverySuggestion" : "Xcode will continue when iPad is connected and unlocked.",
      "domain" : "com.apple.platform.iphoneos"
    },
    "operatingSystemVersion" : "16.1.1 (20B101)",
    "identifier" : "xxx",
    "platform" : "com.apple.platform.iphoneos",
    "architecture" : "arm64e",
    "interface" : "usb",
    "available" : false,
    "name" : "iPad",
    "modelUTI" : "com.apple.ipad-pro-11-4th-2"
  },
  ```

Reply:
Ahhh, that makes sense. Could you add a dartdoc to this field, noting its initial value from xcdevice list could be wrong?